task.sql

```diff
@@ -1,11 +1,24 @@
--- Use our database
-USE ShopDB; 
+DELIMITER $$
+USE ShopDB;
+CREATE PROCEDURE CreateOrder()
+BEGIN
 
--- Some data should be created outside the transaction (here)
+    DECLARE last_order_id INT;
+    DECLARE warehouse_amount INT;
+    INSERT INTO Orders (CustomerID, Date) VALUES (1, '2023-01-02');
+    SET last_order_id = LAST_INSERT_ID();
+    START TRANSACTION;
+    SET warehouse_amount = (SELECT WarehouseAmount FROM Products WHERE ID = 1);
+    IF warehouse_amount >= 1 THEN
+        INSERT INTO OrderItems (OrderID, ProductID, Count)
+        VALUES (@last_order_id, 1, 1);
```

Comment:
The variable `@last_order_id` is used incorrectly. It should be `last_order_id` without the `@` symbol, as it is declared using the `DECLARE` statement and not as a session variable.